plugins/aws/es/esAccessFromIps.js

```diff
@@ -0,0 +1,112 @@
+            default: ''
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings);
+
+        var whitelisted_ip_addresses = settings.whitelisted_ip_addresses || this.settings.whitelisted_ip_addresses.default;
+
+        if (!whitelisted_ip_addresses.length) return callback(null, results, source);
+
+        whitelisted_ip_addresses = whitelisted_ip_addresses.split(',');
+
+        async.each(regions.es, function(region, rcb) {
+            var listDomainNames = helpers.addSource(cache, source,
+                ['es', 'listDomainNames', region]);
+
+            if (!listDomainNames) return rcb();
+
+            if (listDomainNames.err || !listDomainNames.data) {
+                helpers.addResult(
+                    results, 3,
+                    `Unable to query for ES domains: ${helpers.addError(listDomainNames)}`, region);
+                return rcb();
+            }
+
+            if (!listDomainNames.data.length){
+                helpers.addResult(results, 0, 'No ES domains found', region);
+                return rcb();
+            }
+
+            async.each(listDomainNames.data, function(domain, dcb){
+                var describeElasticsearchDomain = helpers.addSource(cache, source,
+                    ['es', 'describeElasticsearchDomain', region, domain.DomainName]);
+
+                if (!describeElasticsearchDomain ||
+                    describeElasticsearchDomain.err ||
+                    !describeElasticsearchDomain.data ||
+                    !describeElasticsearchDomain.data.DomainStatus) {
+                    helpers.addResult(results, 3,
+                        `Unable to query for ES domain config: ${helpers.addError(describeElasticsearchDomain)}`, region);
+                    return dcb();
+                }
+
+                var resource = describeElasticsearchDomain.data.DomainStatus.ARN;
+                
+                if (!describeElasticsearchDomain.data.DomainStatus.AccessPolicies) {
+                    helpers.addResult(results, 0,
+                        'No access policy found', region, resource);
+                    return dcb();
+                }
+
+                var statements = helpers.normalizePolicyDocument(describeElasticsearchDomain.data.DomainStatus.AccessPolicies);
+                var globalAccess = false;
+                var intruderIps = [];
+
+                for (var s in statements) {
+                    var statement = statements[s];
+                    if (!statement.Condition && statement.Principal && helpers.globalPrincipal(statement.Principal)) {
+                        globalAccess = true;
+                        break;
+                    }
+
+                    if (statement.Condition && statement.Condition.IpAddress && statement.Condition.IpAddress['aws:SourceIp']) {
```

Comment:
check the length before looping through this
```suggestion
                    if (statement.Condition && statement.Condition.IpAddress && statement.Condition.IpAddress['aws:SourceIp'] && statement.Condition.IpAddress['aws:SourceIp'].length) {
```